plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableDataStore.php

```diff
@@ -897,4 +1058,107 @@ public function get_database_schema() {
 
 		return $sql;
 	}
+
+	/**
+	 * Returns an array of meta for an object.
+	 *
+	 * @param  WC_Data $object WC_Data object.
+	 * @return array
+	 */
+	public function read_meta( &$object ) {
+		return $this->data_store_meta->read_meta( $object );
+	}
+
+	/**
+	 * Deletes meta based on meta ID.
+	 *
+	 * @param  WC_Data  $object WC_Data object.
+	 * @param  stdClass $meta (containing at least ->id).
+	 */
+	public function delete_meta( &$object, $meta ) {
+		return $this->data_store_meta->delete_meta( $object, $meta );
+	}
+
+	/**
+	 * Add new piece of meta.
+	 *
+	 * @param  WC_Data  $object WC_Data object.
+	 * @param  stdClass $meta (containing ->key and ->value).
+	 * @return int meta ID
+	 */
+	public function add_meta( &$object, $meta ) {
+		return $this->data_store_meta->add_meta( $object, $meta );
```

Comment:
this will insert the meta immediately, shouldn't we do it on save?

Reply:
I agree with you, but this is how historically datastores have worked. There is even the expectation that this call would return the meta ID.
That said, `add_meta()` is called by `update()` so it effectively inserts metadata during save... except if you manually call the datastore method from the outside, which is probably not the most common scenario. wdyt?